feat: add macros for `u_u`, `t_t` and `b_b`

gunjjoshi opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: passed
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds the following 3 macros:
    -   `STDLIB_MATH_BASE_NAPI_MODULE_B_B`
    -   `STDLIB_MATH_BASE_NAPI_MODULE_T_T`
    -   `STDLIB_MATH_BASE_NAPI_MODULE_U_U`

## Related Issues

> Does this pull request have any related issues?

None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gunjjoshi committed:
feat: add macros for u_u, t_t and b_b

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/napi/unary/README.md:
````diff
@@ -610,6 +673,69 @@ The function accepts the following arguments:
 void stdlib_math_base_napi_i_f( napi_env env, napi_callback_info info, float (*fcn)( int32_t ) );
 ```
 
+#### STDLIB_MATH_BASE_NAPI_MODULE_T_T( fcn )
````
@gunjjoshi As per L107, this macro and the function below are not in alphabetical order.

kgryte replied:
Both of these should go after `I_I`.

kgryte commented on lib/node_modules/@stdlib/math/base/napi/unary/include/stdlib/math/base/napi/unary/b_b.h:
```diff
@@ -0,0 +1,85 @@
+* // Register a Node-API module:
+* STDLIB_MATH_BASE_NAPI_MODULE_B_B( scale );
+*/
+#define STDLIB_MATH_BASE_NAPI_MODULE_B_B( fcn )                                \
```
@gunjjoshi All of the `\` should be column-aligned. Ideally, this is at the 80-char column limit.

kgryte commented on lib/node_modules/@stdlib/math/base/napi/unary/include/stdlib/math/base/napi/unary/u_u.h:
```diff
@@ -0,0 +1,84 @@
+* // Register a Node-API module:
+* STDLIB_MATH_BASE_NAPI_MODULE_U_U( scale );
+*/
+#define STDLIB_MATH_BASE_NAPI_MODULE_U_U( fcn )                                \
```
The `\` should be aligned here, as well.

kgryte commented on lib/node_modules/@stdlib/math/base/napi/unary/src/b_b.c:
```diff
@@ -0,0 +1,71 @@
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @param fcn    unary function
+* @return       function return value as a Node-API unsigned 8-bit integer
```
```suggestion
* @return       function return value as a Node-API unsigned 32-bit integer
```

kgryte commented on lib/node_modules/@stdlib/math/base/napi/unary/src/t_t.c:
```diff
@@ -0,0 +1,70 @@
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @param fcn    unary function
+* @return       function return value as a Node-API unsigned 16-bit integer
```
```suggestion
* @return       function return value as a Node-API unsigned 32-bit integer
```

kgryte commented on lib/node_modules/@stdlib/math/base/napi/unary/include/stdlib/math/base/napi/unary/t_t.h:
```diff
@@ -0,0 +1,85 @@
+* // Register a Node-API module:
+* STDLIB_MATH_BASE_NAPI_MODULE_T_T( scale );
+*/
+#define STDLIB_MATH_BASE_NAPI_MODULE_T_T( fcn )                                \
```
The `\` should be aligned here as well.

kgryte reviewed (changes requested):
Left an initial round of comments.

gunjjoshi committed:
docs: align slashes, follow alphabetical order

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/napi/unary/src/b_b.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/napi/unary/src/t_t.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi commented:
Thanks for the review, I've made the requested changes.

kgryte reviewed (approved):
LGTM
